Lesson2/SimpleApp/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SimpleApp.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SimpleApp.Controllers
+{
+    public class ViewProductResult<TProduct>
+    {
+        public IEnumerable<TProduct> ProductsList { get; set; }
+        public string CategoryName { get; set; }
+        public ViewProductResult(IEnumerable<TProduct> productsList, string categoryName)
+        {
+            ProductsList = productsList;
+            CategoryName = categoryName;
+        }
+    }
+
+
+    public class ProductsController : Controller
+    {
+        private readonly ProductReader _reader;
+
+        public ProductsController()
+        {
+            _reader = new ProductReader();
```

Comment:
try to do it with dependency injection: 
add parameter to ctor - `public ProductsController(ProductReader reader)`
register ProductReader in Startup.cs `ConfigureServices` method